src/network/netdev/wireguard.c

```diff
@@ -846,38 +846,35 @@ static void wireguard_done(NetDev *netdev) {
         set_free(w->peers_with_failed_endpoint);
 }
 
-static int wireguard_read_private_key_file(Wireguard *w, bool fatal) {
+static int wireguard_read_private_key_file(Wireguard *w) {
         _cleanup_free_ char *contents = NULL;
         _cleanup_free_ void *key = NULL;
         size_t size, key_len;
         NetDev *netdev;
-        int level, r;
+        int r;
 
         assert(w);
-
         netdev = NETDEV(w);
+        assert(netdev);
 
         if (!w->private_key_file)
                 return 0;
 
-        level = fatal ? LOG_ERR : LOG_INFO;
-
         r = read_full_file(w->private_key_file, &contents, &size);
```

Comment:
(i wonder if we should add a read_fully_file() flavour of some form that warns about files that are world readable for cases like this)